avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -590,7 +628,26 @@ private static Object maybeWrapSchemaless(Schema schema, Object value, String ty
         baseSchema = org.apache.avro.SchemaBuilder.builder().booleanType();
         break;
       case STRING:
-        baseSchema = org.apache.avro.SchemaBuilder.builder().stringType();
+        if (schema.parameters() != null && schema.parameters().containsKey(AVRO_TYPE_ENUM)){
+
+          String enumSchemaName = schema.parameters().get(AVRO_TYPE_ENUM);
```

Comment:
If the name -> full name change causes compatibility issues then we'd want it behind a feature/config flag. (We'd also like to keep these to a minimum, so if there are multiple things that need to be hidden behind a flag, we'd probably want just one flag for all of those so we minimize confusion for users and make maintenance simpler.)